healthcheck/ticker.go

```diff
@@ -1,35 +1,135 @@
 package healthcheck
 
 import (
-	"github.com/ONSdigital/go-ns/log"
+	"sync"
 	"time"
+
+	"github.com/ONSdigital/go-ns/log"
 )
 
 // Ticker calls the health check monitor function at configured intervals.
 type Ticker struct {
 	timeTicker *time.Ticker
+	asyncCheck chan bool
+	closing    chan bool
+}
+
+// NewTicker returns a new ticker that checks the given clients at intervals of the given duration
+func NewTicker(duration, recoveryDuration time.Duration, clients ...Client) *Ticker {
+	return NewTickerWithAlerts(duration, recoveryDuration, nil, nil, clients...)
 }
 
-// NewTicker returns a new ticker that checks the given clients at intervals of the given duration.
-func NewTicker(duration time.Duration, clients ...Client) *Ticker {
+// NewTickerWithAlerts returns a new ticker that checks the given clients at intervals
+// - intervals vary in length (when healthy, use `duration`; at startup, or unhealthy state, use `recoveryDuration`)
+// - sends alerts of any change in health via the given channel (if non-nil) - with the new state (i.e. `true` if newly-healthy)
```

Comment:
I assume it will be false if it's newly not healthy? might be worth explicitly stating this

Reply:
you are right, I will add clarification